### PR TITLE
feat(repo): stabilize companion history and availability flows across web and mobile

### PR DESCRIPTION
**Pull request by ghost**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/YosemiteCrew/Yosemite-Crew/blob/main/CONTRIBUTING.md#commit.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [ ] All existing tests and lints pass.

## What is the current behavior?
Companion history behavior is inconsistent across key frontend surfaces, availability logic has interval/slot edge cases, and integration/permission routing is not fully aligned in all paths.

Mobile invoice/app update policy behavior and Merck manual search flow also needed tighter alignment with appointment-linked UX.

## What is the new behavior?
This PR aligns behavior across affected domains in one pass:

1. Frontend companion history is stabilized across companion detail/list/timeline views.
2. Availability interval and slot normalization is consolidated and applied in appointment flows.
3. Integration and route-permission behavior is hardened for more predictable access handling.
4. Mobile appointment invoice handling and app-update policy flow are aligned with expected behavior.
5. Merck mobile flow (`apps/mobileAppYC/src/features/merck/`) is updated to match appointment context and reduce UX inconsistencies.
6. Repo-level drafting guidance is updated to enforce diff-driven Issue/PR writing for broad cross-workspace changes.

## Related Issue(s)
Fixes #
#1464 

## BREAKING CHANGES:
NONE

### CODE REVIEW

**Comment by ghost** [comment]
codex

File: apps/mobileAppYC/src/features/appUpdate/services/appUpdatePolicy.ts [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+  if (!mustUpdate && !shouldSuggestUpdate) {
+    return null;
+  }
+
+  return {
+    kind: mustUpdate ? 'required' : 'optional',
+    title: platformPolicy?.title ?? appUpdate.title,
+    message: platformPolicy?.message ?? appUpdate.message,
+    storeUrl: resolveStoreUrl(appUpdate, bundleId),
```
[the remark this answers is not in the document]
Comment (ghost) [reply apps/mobileAppYC/src/features/appUpdate/services/appUpdatePolicy.ts]: fixed in latest comit

File: apps/mobileAppYC/src/features/account/services/profileService.ts [hunk 2]
```diff
@@ -187,107 +177,32 @@ const mapSummaryFromResource = (
...
-    address: buildAddressPayload(payload.address),
-    photo: buildPhotoPayload(payload),
-    birthDate: payload.dateOfBirth ?? undefined,
-    extension: extensions.length ? extensions : undefined,
-  };
+    firstName: payload.firstName,
+    lastName: payload.lastName,
+    birthDate: payload.dateOfBirth ? new Date(payload.dateOfBirth) : undefined,
+    email: payload.email,
+    phoneNumber: payload.phoneNumber,
```
[the remark this answers is not in the document]
Comment (ghost) [reply apps/mobileAppYC/src/features/account/services/profileService.ts:189]: fixed in latest comit

**Comment by ghost** [comment]
codex

File: apps/frontend/src/app/features/companionHistory/pages/CompanionHistoryPage.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import { useCompanionTerminologyText } from '@/app/hooks/useCompanionTerminologyText';
+import { getSafeImageUrl, ImageType } from '@/app/lib/urls';
+import Image from 'next/image';
+
+const FALLBACK_BACK_PATH = '/companions';
+const APPOINTMENTS_BACK_PATH = '/appointments';
+
+const resolveSafeBackPath = (candidate: string | null, source: string | null): string => {
+  if (candidate?.startsWith('/')) {
+    return candidate;
```
[the remark this answers is not in the document]
Comment (ghost) [reply apps/frontend/src/app/features/companionHistory/pages/CompanionHistoryPage.tsx]: fixed in latest comit

File: apps/mobileAppYC/src/features/payments/screens/PaymentInvoiceScreen.tsx [hunk 4]
```diff
@@ -357,6 +383,25 @@ const resolveInvoicePaymentStatusLabel = (invoice: Invoice | null) => {
...
+      padding="4"
+      shadow="base"
+      style={styles.glassCard}
+      fallbackStyle={styles.cardFallback}>
+      <View style={styles.cardContent}>
+        <Text style={styles.metaTitle}>Important</Text>
+        <Text style={styles.termsLine}>
+          This appointment was paid in cash and has been cancelled. If a refund
+          is needed, it must be handled directly by the service provider.
+        </Text>
```
[the remark this answers is not in the document]
Comment (ghost) [reply apps/mobileAppYC/src/features/payments/screens/PaymentInvoiceScreen.tsx]: fixed in latest comit

**Review by harshvardhan-yc — APPROVED** [review]
Codex checks are clear. Approving it
